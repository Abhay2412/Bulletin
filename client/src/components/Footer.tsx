// IMPORTING APIS
import React from "react";
import { AppBar, Toolbar, IconButton, useMediaQuery, Button, Menu, MenuItem, ListItemIcon} from "@material-ui/core";
import Typography from '@mui/material/Typography';
import MenuIcon from '@mui/icons-material/Menu';
import { makeStyles, useTheme } from "@material-ui/core/styles";
import { BrowserRouter, Route, Routes, Link } from "react-router-dom";
// REACT APP IMPORTS
import LogIn from "../pages/Login";
import SignUn from "../pages/Signup";
//The local styling used here 
const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  },
  tabs: {
    width: '100%'
  }
}));

const Footer = () => (
    <div className="footer">
      <p>Footer Content will go in here</p>
    </div>
  );
  
  export default Footer;