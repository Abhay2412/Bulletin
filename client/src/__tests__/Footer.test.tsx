import React from "react";
import { configure, mount, ReactWrapper, shallow } from "enzyme";
import { AppBar, Toolbar, Button, IconButton } from "@material-ui/core";
import Footer from "../components/Footer";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("testing <Footer /> component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <Footer />
    </Provider>
  );

  it("should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Footer />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render one Footer component", () => {
    expect(wrapper.find(AppBar).length).toEqual(1);
  });

  it("should render Footer with static poistioning", () => {
    expect(wrapper.find(AppBar).prop("position")).toEqual("static");
  });

  it("should render exactly three <Button /> component", () => {
    expect(wrapper.find(Button).length).toEqual(3);
  });

  it("should render exactly 1 <Toolbar /> component", () => {
    expect(wrapper.find(Toolbar).length).toEqual(1);
  });
});
