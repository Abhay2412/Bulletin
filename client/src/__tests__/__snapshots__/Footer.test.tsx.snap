// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Footer /> component should render properly 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    style={
      Object {
        "background": "#F0E6DB",
      }
    }
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap makeStyles-title-3 css-u2cfxh-MuiTypography-root"
      >
        BULLETIN
        <img
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTBtsNMynXKIdCnskttK4XjgIDre85wUEi_8hCjlXqnJIiMlKVJhCtzOPM-cWKfyFjeMmY&usqp=CAU"
          width="39"
        />
      </div>
      <div
        style={
          Object {
            "marginRight": "2rem",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Connect With Us
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Office Location
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Company
          </span>
        </button>
      </div>
    </div>
  </header>
</div>
`;
